app/features/worksheet_generator/tools.py

```diff
@@ -0,0 +1,292 @@
+from langchain_core.runnables import RunnablePassthrough, RunnableParallel
+from langchain_community.vectorstores import Chroma
+from langchain.text_splitter import CharacterTextSplitter
+from langchain_community.document_loaders import (
+    PyPDFLoader,
+    UnstructuredWordDocumentLoader,
+    UnstructuredExcelLoader,
+    UnstructuredHTMLLoader,
+    UnstructuredPowerPointLoader,
+    UnstructuredCSVLoader,
+)
+from langchain_core.documents import Document
+import google.auth
+from PIL import Image
+import pytesseract
+import torchaudio
+from moviepy.editor import VideoFileClip
+import speech_recognition as sr
+#print("SpeechRecognition imported successfully")
+
+from pydub import AudioSegment
+
+# Set the path to the service account key
+os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = "/home/sanjana3014/kai-ai-backend_/local-auth.json"  # Ensure this is the correct path
+
+
+class DocumentProcessor:
+    def __init__(self):
+        self.pages = []  # List to keep track of pages from all documents
+
+    def read_image(self, file_path):
+        image = Image.open(file_path)
+        text = pytesseract.image_to_string(image)
+        return text
+
+    def read_audio(self, file_path):
+        if file_path.endswith(".mp3"):
+            audio = AudioSegment.from_mp3(file_path)
+            wav_path = file_path.replace(".mp3", ".wav")
+            audio.export(wav_path, format="wav")
+            file_path = wav_path
+        recognizer = sr.Recognizer()
+        with sr.AudioFile(file_path) as source:
+            audio = recognizer.record(source)
+        try:
+            text = recognizer.recognize_google(audio)
+            return text
+        except sr.UnknownValueError:
+            return "Could not understand the audio"
+        except sr.RequestError as e:
+            return f"Could not request results; {e}"
+
+    def read_video(self, file_path):
+        video = VideoFileClip(file_path)
+        audio_path = os.path.join(tempfile.gettempdir(), f"{uuid.uuid4().hex}.wav")
+        video.audio.write_audiofile(audio_path)
+        text = self.read_audio(audio_path)
+        os.remove(audio_path)
+        return text
+    
+    def ingest_documents(self):
+        uploaded_files = st.file_uploader("Upload files", type=['pdf', 'png', 'mp4', 'docx', 'xlsx', 'csv', 'pptx', 'html', 'jpg', 'jpeg', 'wav', 'mp3'], accept_multiple_files=True)
+        
+        if uploaded_files:
+            for uploaded_file in uploaded_files:
+                unique_id = uuid.uuid4().hex
+                original_name, file_extension = os.path.splitext(uploaded_file.name)
+                temp_file_name = f"{original_name}_{unique_id}{file_extension}"
+                temp_file_path = os.path.join(tempfile.gettempdir(), temp_file_name)
+
+                with open(temp_file_path, 'wb') as f:
+                    f.write(uploaded_file.getvalue())
+
+                if file_extension == ".pdf":
+                    loader = PyPDFLoader(temp_file_path)
```

Comment:
As @mikhailocampo mentioned, you can use enums for better scalability and reuse these components. You can try creating a reusable function for this.